pyccel/ast/bind_c.py

```diff
@@ -114,21 +115,61 @@ def as_static_function(func, name=None):
                         )
 
 #=======================================================================================
-def as_static_function_call(func, mod_name, name=None):
+def as_static_module(funcs, original_module, name = None):
+    """ Create the module contained in the bind_c_mod.f90 file
+    This is the interface between the c code and the fortran code thanks
+    to iso_c_bindings
+
+    Parameters
+    ==========
+    funcs : list of FunctionDef
+            All the functions which may be exposed to c
+    original_module : str
+            The name of the module being wrapped
+    name  : str
+            The name of the new module
+    """
+    funcs = [f for f in funcs if not f.is_private]
+    imports = []
+    bind_c_funcs = [as_static_function_call(f, original_module, imports = imports) for f in funcs]
+    if name is None:
+        name = 'bind_c_{}'.format(original_module)
+    return Module(name, (), bind_c_funcs, imports = imports)
+
+#=======================================================================================
+def as_static_function_call(func, mod_name, name=None, imports = None):
+    """ Translate a FunctionDef to a BindCFunctionDef which calls the
+    original function. A BindCFunctionDef is a FunctionDef where the
+    arguments are altered to allow the function to be called from c.
+    E.g. the size of each dimension of an array is provided
+
+    Parameters
+    ==========
+    func     : FunctionDef
+               The function to be translated
+    mod_name : str
+               The name of the module which contains func
+    name     : str
+               The new name of the function
+    imports  : list
+               An optional parameter into which any required imports
+               can be collected
```

Comment:
If `imports` is not passed this function will behave (almost) as before?

Reply:
I've made a quick search, and this function is called without the `imports` argument in `cwrappercode.py`. Can you have a look at it?

Reply:
It's just used to get the function signature to be able to print it at the top of the `X_wrapper.c` file. It's definitely overkill to create a full `FunctionDef` just for that though, we should be calling `as_static_function` or better `as_static_function_signature`

Reply:
Actually the cwrapper really ought to just operate on the created module rather than calling these functions again

Reply:
I'm not sure this is worth correcting now thought as @MenoIy is rewriting lots of the cwrapper. I suspect it will be easier to do this afterwards

Reply:
The implementation is not trivial in the case of static functions if we want to avoid repetition. I think I will add a `as_static_function_signature` for now and we can always improve it later

Reply:
See issue #801

Reply:
Thanks!